servlet/src/main/java/io/undertow/servlet/spec/ServletPrintWriter.java

```diff
@@ -254,45 +236,84 @@ public void write(final char[] buf, final int off, final int len) {
         write(cb);
     }
 
+    private static int writeAndFlushAscii(ServletOutputStreamImpl outputStream, ByteBuffer buffer, char[] chars, int start, int end) throws IOException {
+        assert buffer.order() == ByteOrder.BIG_ENDIAN;
+        int i = start;
```

Comment:
So I'm curious: why is the buffer big endian?

Reply:
I have noticed that the existing code doesn't change it from the default endianness of NIO, but if your question is re "why it was like that?" I have my own theories:
1. By design: existing framing protocol uses big endian buffers for some by-design reason
2. Historical: networking often makes uses of big endian buffers 
3. Uups: NIO buffers while allocated are big endian due to historical reasons AFAIK, and the existing code doesn't enforce native and/or little endianess because the http txt protocol doesn't make use of it 

I am keen to vote for the third option @stuartwdouglas probably knows